docs/tutorials/infrastructure/kne/srl-with-kne/topology.md

```diff
@@ -0,0 +1,193 @@
+nodes: {
+    // first node parameters
+}
+
+nodes: {
+    // second node parameters
+}
+```
+
+Node definition has quite some parameters[^2]. We will cover the most common of them.
+
+### Name
+
+Each node must have a name, which is a free-formed string:
+
+```proto
+nodes: {
+    name: "srl1"
+    // other node parameters snipped for brevity
+}
+```
+
+### Vendor
+
+To let KNE know which vendor is defined within the node section, the `vendor` field is provided. Several vendors are supported by KNE and the full list is provided in the [topo.proto](https://github.com/openconfig/kne/blob/9d835bbaa1e4b26ab03b0d456461a044f2ec9ea0/proto/topo.proto#L30) file.
+
+```proto
+nodes: {
+    vendor: NOKIA # (1)!
+    // other node parameters snipped for brevity
+}
+```
+
+1. Note, the vendor value needs to be provided exactly as defined in the Vendor enum field of the proto file. Without quotes.
+
+!!!note
+    Some KNE examples may utilize `type` parameter with a value of `NOKIA_SRL` or similar. This filed is going to be deprecated in favor of seperate fields: `vendor`/`model`/`os`/`version`.
+
+When working with Nokia SR Linux nodes, the vendor field must be set to `NOKIA`.
+
+### Model
+
+With the `model` field a user indicates which particular model of a given Vendor should be used by the node. In the context of Nokia SR Linux, the `model` field drives the [hardware variant](../../../../kb/hwtypes.md) that an SR Linux container will emulate.
+
+```proto
+nodes: 
+    model: "ixrd3l"
+    // other node parameters snipped for brevity
+}
+```
+
+### Config
+
+Parameters that configure the way a k8s pod representing a network node is deployed are grouped under the [Config](https://github.com/openconfig/kne/blob/9d835bbaa1e4b26ab03b0d456461a044f2ec9ea0/proto/topo.proto#L112) message.
+
+#### Image
+
+The essential field under the Config block is `image`. It sets the container image name that k8s will use in the pod specification.
+
+```proto
+nodes:
+    config:{
+        image: "ghcr.io/nokia/srlinux:22.6.1"
+    }
+    // other node parameters snipped for brevity
+}
+```
+
+!!!note
+    When `kind` cluster is used, users might want to [load the container image](installation.md#image-load) before creating the topologies.
+
+#### File
+
+Often it is desired to deploy a node with a specific startup configuration applied. KNE enables this use case by using the `file` parameter of a `Config` message. A path to the startup configuration file is provided using the path relative to the topology file.
+
+```proto
+nodes:
+    config:{
+        file: "my-startup-config.json"
+    }
+    // other node parameters snipped for brevity
+```
+
+In the snippet above, the `my-stratup-config.json` is expected to be found next to the topology file.
+
+For SR Linux nodes, the startup file must contain the full configuration of a node in a JSON format as found in `/etc/opt/srlinux/config.json` on SR Linux filesystem.
+
+### Services
```

Comment:
maybe mention the cert generation somewhere in here (proto vs config)

Reply:
done in [c266bdc](https://github.com/srl-labs/learn-srlinux/pull/38/commits/c266bdc1acb400fdadfe84772962523b8b4ea7ef)